src/java.base/share/classes/java/lang/StringTemplate.java

```diff
@@ -0,0 +1,612 @@
+        Objects.requireNonNull(values, "values must not be null");
+        int fragmentsSize = fragments.size();
+        int valuesSize = values.size();
+        if (fragmentsSize != valuesSize + 1) {
+            throw new IllegalArgumentException("fragments must have one more element than values");
+        }
+        JavaTemplateAccess JTA = SharedSecrets.getJavaTemplateAccess();
+        return JTA.interpolate(fragments, values);
+    }
+
+    /**
+     * Combine zero or more {@link StringTemplate StringTemplates} into a single
+     * {@link StringTemplate}.
+     * {@snippet :
+     * StringTemplate st = StringTemplate.combine(RAW."\{a}", RAW."\{b}", RAW."\{c}");
+     * assert st.interpolate().equals(STR."\{a}\{b}\{c}");
+     * }
+     * Fragment lists from the {@link StringTemplate StringTemplates} are combined end to
+     * end with the last fragment from each {@link StringTemplate} concatenated with the
+     * first fragment of the next. To demonstrate, if we were to take two strings and we
+     * combined them as follows: {@snippet lang = "java":
+     * String s1 = "abc";
+     * String s2 = "xyz";
+     * String sc = s1 + s2;
+     * assert Objects.equals(sc, "abcxyz");
+     * }
+     * the last character {@code "c"} from the first string is juxtaposed with the first
+     * character {@code "x"} of the second string. The same would be true of combining
+     * {@link StringTemplate StringTemplates}.
+     * {@snippet lang ="java":
+     * StringTemplate st1 = RAW."a\{}b\{}c";
+     * StringTemplate st2 = RAW."x\{}y\{}z";
+     * StringTemplate st3 = RAW."a\{}b\{}cx\{}y\{}z";
+     * StringTemplate stc = StringTemplate.combine(st1, st2);
+     *
+     * assert Objects.equals(st1.fragments(), List.of("a", "b", "c"));
+     * assert Objects.equals(st2.fragments(), List.of("x", "y", "z"));
+     * assert Objects.equals(st3.fragments(), List.of("a", "b", "cx", "y", "z"));
+     * assert Objects.equals(stc.fragments(), List.of("a", "b", "cx", "y", "z"));
+     * }
+     * Values lists are simply concatenated to produce a single values list.
+     * The result is a well-formed {@link StringTemplate} with n+1 fragments and n values, where
+     * n is the total of number of values across all the supplied
+     * {@link StringTemplate StringTemplates}.
+     *
+     * @param stringTemplates  zero or more {@link StringTemplate}
+     *
+     * @return combined {@link StringTemplate}
+     *
+     * @throws NullPointerException if stringTemplates is null or if any of the
+     * {@code stringTemplates} are null
+     *
+     * @implNote If zero {@link StringTemplate} arguments are provided then a
+     * {@link StringTemplate} with an empty fragment and no values is returned, as if invoking
+     * <code>StringTemplate.of("")</code> . If only one {@link StringTemplate} argument is provided
+     * then it is returned unchanged.
+     */
+    static StringTemplate combine(StringTemplate... stringTemplates) {
```

Comment:
Sorry for a late mention, but I believe this method and its List variant are better named `concat` instead: `concat` indicates an explicitly ordered operation (String, Stream, AffineTransform), while `combine` is usually unordered (IntSummaryStatistics, CompletableFuture.thenCombine, Collector.combiner). Using `concat` also brings `String` and `StringTemplate` closer.

All examples come from using the search box with `concat` and `combine` in the code review Javadoc copy: https://cr.openjdk.org/~jlaskey/templates/docs/api/java.base/java/util/FormatProcessor.html